src/core/string_set_test.cc

```diff
@@ -657,4 +657,50 @@ void BM_Grow(benchmark::State& state) {
 }
 BENCHMARK(BM_Grow);
 
+unsigned total_wasted_memory = 0;
+
+TEST_F(StringSetTest, ReallocIfNeeded) {
+  auto build_str = [](size_t i) { return to_string(i) + string(131, 'a'); };
+
+  auto count_waste = [](const mi_heap_t* heap, const mi_heap_area_t* area, void* block,
+                        size_t block_size, void* arg) {
+    size_t used = block_size * area->used;
+    total_wasted_memory += area->committed - used;
+    return true;
+  };
+
+  for (size_t i = 0; i < 10'000; i++)
+    ss_->Add(build_str(i));
```

Comment:
Would maybe worth to add a unit with `ttl` to be 100% sure we copy it properly. So two tests in total:

1. As is now, without ttl
2. With ttl